internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -112,10 +116,27 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 	// NOTE: Once we reconcile node labels for the first time, the NodeUninitializedTaint is removed from the node.
 	nodeLabels := getManagedLabels(machine.Labels)
 
+	// Get interruptible instance status from the infrastructure provider and set the interruptible label on the node.
+	interruptible := false
+	if infra != nil {
+		interruptible, _, err = unstructured.NestedBool(infra.Object, "status", "interruptible")
+		if err != nil {
+			log.V(1).Error(err, "Failed to get interruptible status from infrastructure provider")
+		}
+		// If interruptible is true add the interruptible label to the node labels.
+		if interruptible {
+			nodeLabels[clusterv1.InterruptibleLabel] = ""
+		}
+	}
+
 	// Reconcile node taints
 	if err := r.patchNode(ctx, remoteClient, node, nodeLabels, nodeAnnotations); err != nil {
 		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile Node %s", klog.KObj(node))
 	}
+	if interruptible {
+		log.V(3).Info("Set interruptible label on Machine's Node %s", klog.KObj(node))
+		r.recorder.Event(machine, corev1.EventTypeNormal, "SuccessfulSetInterruptibleNodeLabel", node.Name)
```

Comment:
I think it would be better if we send this event only if the label was not set before. Otherwise we will send it on every reconcile (like the old implementation).

Could we check if node before l.133 contains the label already and only send this event if we add it in this reconcile?

I would get rid of the log. We don't log any of the other labels that we set